src/cadence/mainnet/getNFTs.cdc

```diff
@@ -1248,4 +1250,45 @@ pub fun getXtinglesNFT(owner: PublicAccount, id: UInt64): NFTData? {
             "edition": nft!.metadata!.edition   
         },
     )
-}
\ No newline at end of file
+}
+
+// https://flow-view-source.com/mainnet/account/0x6831760534292098/contract/OneFootballCollectible
+pub fun getOneFootballCollectible(owner: PublicAccount, id: UInt64): NFTData? {
+    let contract = NFTContract(
+        name: "OneFootballCollectible",
+        address: 0x6831760534292098,
+        storage_path: "OneFootballCollectible.CollectionStoragePath",
+        public_path: "OneFootballCollectible.CollectionPublicPath",
+        public_collection_name: "OneFootballCollectible.OneFootballCollectibleCollectionPublic",
+        external_domain: "http://xmas.onefootball.com/"
+    )
+
+    if let collection = owner
+        .getCapability<&OneFootballCollectible.Collection{OneFootballCollectible.OneFootballCollectibleCollectionPublic}>(OneFootballCollectible.CollectionPublicPath)
+            .borrow() {
+        if let nft = collection.borrowOneFootballCollectible(id: id) {
+            if let metadata = nft.getTemplate() {
+                return NFTData(
+                    contract: contract,
+                    id: nft.id,
+                    uuid: nft.uuid,
+                    title: metadata.name,
+                    description: metadata.description,
+                    external_domain_view_url: "https://xmas.onefootball.com/".concat(owner.address.toString()),
+                    token_uri: nil,
+                    media: [
+                        // media
+                        NFTMedia(uri: "https://".concat(metadata.media).concat(".ipfs.dweb.link/"), mimetype: "video"),
+                        // preview
+                        NFTMedia(uri: "https://".concat(metadata.preview).concat(".ipfs.dweb.link/"), mimetype: "image")
```

Comment:
also I'm not sure about adding the preview as fallback, as I don't see anyone doing this in this script

Reply:
This looks good